test/cli/dedup_loc/test.out

```diff
@@ -5,7 +5,4 @@ test/cli/dedup_loc/dedup_loc.rb:5: Argument does not have asserted type `NilClas
     test/cli/dedup_loc/dedup_loc.rb:4:
      4 |rescue Exception => e
         ^^^^^^
-    test/cli/dedup_loc/dedup_loc.rb:4:
-     4 |rescue Exception => e
-               ^^^^^^^^^
```

Comment:
trying to understand this. on your branch:

<img width="408" alt="Screenshot 2023-04-12 at 8 14 44 PM" src="https://user-images.githubusercontent.com/5544532/231639538-7fe6987b-6860-4948-91d0-0495315ab19f.png">

super weird. it appears that this only happens when you put `Exception => e` specifically? if that's the case honestly i'm fine with that, but I wonder if the fact that `foo.rb` in my example above behaves differently from dedup_loc.rb gives you any ideas for how to adjust the loc

I will note that the loc I chose for the `GetCurrentException` instruction is a hack anyways. We _may_ want to put it back to the whole `rescue...end` block? i.e. revert this commit:

https://github.com/sorbet/sorbet/pull/6908/commits/3ea6ff854d7ca41b54185740185d2fd65cf9ab2d

That would at least make it clear that it's the "whole rescue block" participating in the source of `T.nilable(Exception)`, instead of somewhat misleadingly pointing to the first `rescue` clause, which may not mention `Exception`.

Anyways, whacky stuff, thanks for mentioning something in the PR description.

(this is non blocking tbc)